src/libraries/System.Net.HttpListener/src/System/Net/Managed/HttpListenerRequest.Managed.cs

```diff
@@ -139,12 +139,12 @@ private static bool MaybeUri(string s)
             if (p >= 10)
                 return false;
 
-            return IsPredefinedScheme(s.Substring(0, p));
+            return IsPredefinedScheme(s.AsSpan(0, p));
         }
 
-        private static bool IsPredefinedScheme(string scheme)
+        private static bool IsPredefinedScheme(ReadOnlySpan<char> scheme)
         {
-            if (scheme == null || scheme.Length < 3)
+            if (scheme.Length < 3)
```

Comment:
A lot of the code below is now incorrect.  All of the equality operators are going to do the wrong thing.

Reply:
https://github.com/dotnet/runtime/blob/98c9f2960c18336d6a18f2a83d80271c18d64e54/src/libraries/System.Net.HttpListener/src/System/Net/Managed/HttpListenerRequest.Managed.cs#L183-L186

I don't see how this is supposed to be better than just calling into `Uri` directly.
IMO the right optimization here would be to remove the `MaybeUri` entirely.

Reply:
You mean just to remove `MaybeUri(_rawUrl!.ToLowerInvariant()) &&`?

Reply:
I guess this is the exact same logic that is called in `Uri.TryCreate`
https://github.com/dotnet/runtime/blob/dfd618dc648ba9b11dd0f8034f78113d69f223cd/src/libraries/System.Private.Uri/src/System/Uri.cs#L3573-L3585

Reply:
> You mean just to remove MaybeUri(_rawUrl!.ToLowerInvariant()) &&? (And the MaybeUri body)

Yes, if `Uri.TryCreate` isn't able to perform well here, **it** should be improved instead. Duplicating such logic makes no sense.

Reply:
Some of the failures may be due to the fact Uri accepts implicit file Uris as "absolute", so just the path may be accepted as absolute.
Could be mitigated by replacing `MaybeUri` with `StartsWith("http", OrdinalIgnoreCase)` or just reverting the change.